tests/e2e/csi_static_provisioning_basic.go

```diff
@@ -1210,4 +1247,161 @@ var _ = ginkgo.Describe("Basic Static Provisioning", func() {
+				err = client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, metav1.DeleteOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}()
+
+			vmdk := GetAndExpectStringEnvVar(envVmdkDiskURL)
+			framework.Logf("VMDK path : %s", vmdk)
+			ginkgo.By("Create CNS register volume with VMDK")
+			cnsRegisterVolume := getCNSRegisterVolumeSpec(ctx, namespace, "", vmdk, pvcName, v1.ReadWriteOnce)
+			err = createCNSRegisterVolume(ctx, restConfig, cnsRegisterVolume)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			log.Infof("waiting for some time for FCD to register in CNS and for cnsRegisterVolume to get create")
+			//TODO add polling wait
+			time.Sleep(time.Duration(100) * time.Second)
+			cnsRegisterVolumeName := cnsRegisterVolume.GetName()
+			log.Infof("CNS register volume name : %s", cnsRegisterVolumeName)
+
+			ginkgo.By("verify created PV, PVC and check the bidirectional reference")
+			pvc, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvcName, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			pv := getPvFromClaim(client, namespace, pvcName)
+			verifyBidirectionalReferenceOfPVandPVC(ctx, client, pvc, pv, fcdID)
+
+			//TODO: need to add code to delete VMDK hard disk and to create POD
+
+			defer func() {
+				ginkgo.By("Deleting the PV Claim")
+				framework.ExpectNoError(fpv.DeletePersistentVolumeClaim(client, pvcName, namespace), "Failed to delete PVC", pvcName)
+				pvc = nil
+
+				ginkgo.By("PV will be in released state , hence delete PV explicitly")
+				framework.ExpectNoError(fpv.DeletePersistentVolume(client, pv.GetName()))
+				pv = nil
+
+				ginkgo.By("Verify CRD should be deleted automatically")
+				//TODO add polling wait
+				time.Sleep(time.Duration(40) * time.Second)
+				flag := queryCNSRegisterVolume(ctx, restConfig, cnsRegisterVolumeName, namespace)
+				gomega.Expect(flag).NotTo(gomega.BeTrue())
+
+				ginkgo.By("Delete Resource quota")
+				deleteResourceQuota(client, namespace)
+			}()
+
+		} else {
+			framework.Logf("Skipping this test Since the testbed dont have vSAN datastore - Because for this test uses vSAN default datastore policy")
+		}
+
+	})
+
+	/*
+		Specify VolumeID and DiskURL together and verify the error message
+		Presetup: Create VMDK with valid storage policy.
+
+		Test Steps:
+		1. Create FCD
+		2. Create Resource quota
+		3. Create CNS register volume with above created VMDK and FCDID
+		4. Verify the error message "VolumeID and DiskURLPath cannot be specified together"
+	*/
+	ginkgo.It("[csi-supervisor] Specify VolumeID and DiskURL together and verify the error message", func() {
+
+		var err error
+		ctx, cancel := context.WithCancel(context.Background())
+		log := logger.GetLogger(ctx)
+		defer cancel()
+
+		dataStoreType, err := defaultDatastore.Type(ctx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		log.Infof("Datastore type: %s", dataStoreType)
+		if dataStoreType == "vsan" {
```

Comment:
restructure, same as above
and fix comments given for above tests here as well

Reply:
done